### ISSUE TITLE
Make repo private?

### ISSUE THREAD

**Issue by alexpearce** [alexpearce/CharmProduction#95]
Should we make the CharmProduction repository private, i.e. invisible to anyone but us?

I'm on the fence about it because I _really_ like having the codebase publicly accessible; I've already had a couple of people tell me they've solved a problem by finding some of the code in this repo via Google. I also really like having issues side-by-side with the code (something hosting on the LHCb SVN can't get you).

But, now we're starting to discuss some ‘sensitive’ information, like publishing schedules, binning schemas, and so on, which some LHCb members might not be happy with. I think it is unlikely, but having this repo open does technically allow ‘competitors’ to see what we're up to in more detail than if we only communicated via an `lhcb-` mailing list, for example.

Of course, we could just keep it open, and _maybe_ ‘get caught’ talking openly about things we shouldn't, and get our wrists slapped.

Thoughts?

**Comment by chrisburr** [comment]
Firstly I presume you're aware you can get [5 free private repositories as a student](https://education.github.com/pack)?

My thoughts:
##### We can keep it public
- The consequences if we get bitten might be too significant to ignore despite it being unlikely. But I don't know if this is actually the case.

##### We can make it private
- Marco, Patrick, Sajan and Vava loose access to things like the stripping lines unless they make accounts.
- I need to setup some more ssh keys.
- We could make the repo private for now and make it public after the measurement is published so it can still serve as an archive for people. On the down side some of the issues will no longer be relevant.
- Alternatively it looks like its possible to duplicate a repository with all the discussion so we could have a private copy for current work. We can then copy back to the public one when nothing sensitive remains and if we have any issues that might be interesting to others we can [copy](https://github-issue-mover.appspot.com/) them to the public repository.

##### We can lobby LHCb to get GitHub Enterprise
- They even do academic/non-profit discounts! :smile:

**Comment by alexpearce** [comment]
Yep, got mine :smile: :money_with_wings: 

Other people on the analysis not having acccess is definitely the biggest downside to going private. The stripping lines _should_ be going to SVN soon (in the next day or two), but there's no plan for anything else (we could migrate to the LHCb package structure and the syncronise the two, but that's a pain).

Opening it up after the measurement is published is definitely possible, and I would like to do that if we go private.

If LHCb migrated to GitHub I would eat my (nonexistent) hat, but that would be ideal…

**Comment by dominikmuller** [comment]
What did people solve using our repository? ;-)

We could also start a CERN git project, might be they also provide github like features.
Well, they can get access by creating an account which is not too much to ask I guess.

**Comment by alexpearce** [comment]
Yes, we could [request a new project](https://cernforge.cern.ch/) on the [CERN Jira](http://information-technology.web.cern.ch/services/JIRA-service) instance, which has similar issues and stuff to GitHub. What would put me off is friction, as GH is very lightweight and clean, and I know it. Switching would give us One More Thing to do.

True, it's not a lot of work creating an account, but it's one more thing for them to do (and will inevitably raise the question “why not use SVN?”)

**Comment by chrisburr** [comment]
From what I can quickly tell the [CERN git service](http://git.cern.ch/pubweb/davix.git/tree) is very similar to the svn. In any case I think keeping to GitHub is essential as its setup, working well and we all like it.

**Comment by chrisburr** [comment]
If we go private you will probably need to send Travis an email so we [get more builds](https://education.travis-ci.com/). Even if we ignore the multiple builds per commit there has been over 300 in the last month.

**Comment by alexpearce** [comment]
I'm leaning towards private now, the main reason being that I don't want to be accused of leaking any ‘sensitive’ information. So, I propose

* Making the repository private
* If any of our co-analysts want to see it, we tell them to sign up to GH and we'll grant them access
* Making the repo public after publication (this may need to wait until after the second publication, but we can discuss that after the first)

If both of you agree, I'll flip the switch.

**Comment by chrisburr** [comment]
:+1:

**Comment by chrisburr** [comment]
@dominikmuller?

**Comment by dominikmuller** [comment]
Sorry, thought I said that I agree when we had a Skype discussion. Go ahead and flip the switch. Thanks!
